cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -1844,7 +1852,8 @@ x"
              A -> (): 'a2';
              A -> (): 'a3';
             ",
-            YaccParserErrorKind::DuplicateRule
+            YaccParserErrorKind::DuplicateRule,
+            2
```

Comment:
We really need to test a) that the duplicated rule is `A` b) the locations of the duplicates. [We should also add perhaps rules `B` and `C` between each `A`.] Without that more specific test, we could quite easily break the code and return duplicates that are actually of the original rule.

Reply:
I tried to do this in a way that doesn't require me to go hard code spans individually, and still hopefully be reusable
for other duplicate error kinds in the future.  But I believe this is fixed in https://github.com/softdevteam/grmtools/pull/307/commits/add3e128ad789316d955fe79556ba67b3a2e438e

Reply:
I'm afraid we really do have to check the `Span`s (lots of other grmtools tests do this, for good reason).

Reply:
Yep, it's tedious, but we have to do it, because without that we can't a) be sure there aren't bugs now b) we don't introduce bugs later. That's why there are *lots* of hard-coded `Span` tests already.

Reply:
fixed in https://github.com/softdevteam/grmtools/pull/307/commits/fe8974a76d5e28c4465daae75726630f3aaeb4d6